### PR TITLE
containerapp refactor managed env show/list/delete/create/update

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp env show/list/delete/create/update

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by Greedygre** [comment]
Will rerun all tests related with `--live`

**Comment by Greedygre** [comment]
Hi @zhoxing-ms  @wangzelin007 
The test failed due to following error, could you help to handle it?

======================================================================
ERROR: test_ref_doc_db-up (__main__.IndexRefDocs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/vss/_work/1/s/./scripts/ci/index_ref_doc.py", line 71, in test
    raise e
  File "/mnt/vss/_work/1/s/./scripts/ci/index_ref_doc.py", line 68, in test
    check_call(script_args)
  File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/opt/hostedtoolcache/Python/3.10.12/x64/bin/python', '/mnt/vss/_work/1/s/scripts/refdoc/generate.py', '--extension-file', '/tmp/tmpxsznkbhq/db_up-0.2.7-py2.py3-none-any.whl', '--output-dir', '/tmp/tmp0179mina/db-up']' returned non-zero exit status 1.

----------------------------------------------------------------------

**Comment by wangzelin007** [comment]
> Hi @zhoxing-ms @wangzelin007 The test failed due to following error, could you help to handle it?
> 
> ## ======================================================================
> ERROR: test_ref_doc_db-up (**main**.IndexRefDocs)
> Traceback (most recent call last): File "/mnt/vss/_work/1/s/./scripts/ci/index_ref_doc.py", line 71, in test raise e File "/mnt/vss/_work/1/s/./scripts/ci/index_ref_doc.py", line 68, in test check_call(script_args) File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/subprocess.py", line 369, in check_call raise CalledProcessError(retcode, cmd) subprocess.CalledProcessError: Command '['/opt/hostedtoolcache/Python/3.10.12/x64/bin/python', '/mnt/vss/_work/1/s/scripts/refdoc/generate.py', '--extension-file', '/tmp/tmpxsznkbhq/db_up-0.2.7-py2.py3-none-any.whl', '--output-dir', '/tmp/tmp0179mina/db-up']' returned non-zero exit status 1.

This issue is casued by pymssql library used by the db-up extension, and I created an issue [here](https://github.com/pymssql/pymssql/issues/832) to track it.

File: src/containerapp/azext_containerapp/containerapp_env_decorator.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+        location = self.get_argument_location()
+        if self.get_argument_zone_redundant():
+            if not self.get_argument_infrastructure_subnet_resource_id():
+                raise RequiredArgumentMissingError("Cannot use --zone-redundant/-z without "
+                                                   "--infrastructure-subnet-resource-id/-s")
+            if not is_valid_resource_id(self.get_argument_infrastructure_subnet_resource_id()):
+                raise ValidationError("--infrastructure-subnet-resource-id must be a valid resource id")
+            vnet_location = get_vnet_location(self.cmd, self.get_argument_infrastructure_subnet_resource_id())
+            if location:
+                if _normalize_location(self.cmd, location) != vnet_location:
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/containerapp_env_decorator.py:133]: make sure vnet location noormalized

**Review by Juliehzl — APPROVED** [review]

File: src/containerapp/azext_containerapp/containerapp_env_decorator.py (hunk 1 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/containerapp_env_decorator.py:133]: `get_vnet_location` has been normalize

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
Could you please help to review and merge this PR, thanks a lot!

**Comment by Greedygre** [comment]
> Hi @Greedygre, Please write the description of changes which can be perceived by customers into HISTORY.rst. If you want to release a new extension version, please update the version in setup.py as well.

This pr's changes cannot be perceived by customers. So should not update the history.rst.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**